src/unity/Assets/Crest/Scripts/Shapes/ShapeGerstnerBatched.cs

```diff
@@ -74,14 +81,23 @@ void Update()
 
         void UpdateAmplitudes()
         {
+            _mostSignificant.Clear();
             if (_amplitudes == null || _amplitudes.Length != _wavelengths.Length)
             {
                 _amplitudes = new float[_wavelengths.Length];
             }
-
+            if (_waveSpeed == null || _waveSpeed.Length != _wavelengths.Length)
+            {
+                _waveSpeed = new float[_wavelengths.Length];
+            }
             for (int i = 0; i < _wavelengths.Length; i++)
             {
+                _waveSpeed[i] = ComputeWaveSpeed(_wavelengths[i]);
                 _amplitudes[i] = _spectrum.GetAmplitude(_wavelengths[i]);
+                if (_amplitudes[i] > m_SignificantThresholdForCPU)
+                {
+                    _mostSignificant.Add(i);
```

Comment:
good stuff. not required as part of this review, but an idea - it *might* be fruitful to actually make duplicate versions of _amplitudes, _waveSpeed, _wavelengths that have only the significant components, instead of having an array of indexes. this could be better for cache coherence. however, you might see 0 improvement for it, especially if all the data already fits in a 4k cache line which im sure it does. as i said, not required for this review but thought i'd braindump. i could also give this a stab in the future if useful. (and a random thought that just occurred to me you want to be profiling in fully optimised standalone builds if you're not already - you'll likely see quite different behaviour and bottlenecks).

Reply:
I did consider this, but in my use case the getheight is called high frequency one after the other so the cache should be good . Also the full arrays are small like 256 bytes small. If we really want to go all in, making the array access unsafe will remove the bounds checks that will be happening 

Yeah profiling in standalone is about 3x faster but not exceptionally different from one function to the next, the unity profiler is pretty bad, and absolutely cant be trusted in deep profile mode, then its overhead outweighs any useful data.

Reply:
ah i didnt know the unsafe array trick! thanks for the notes. happy to go with what you have.